CMake/CompilerFlagsHelpers.cmake

```diff
@@ -0,0 +1,118 @@
+#
```

Comment:
Compiler flags portability, nothing more.

Reply:
I don't see a lot of CMake projects using this pattern and changing these default flags. Why is this patch doing it?

Reply:
@EricWF Because a lot of projects do not care to run on platforms with exotic compilers / old compilers that are badly handle by default cmake. I unfortunately have this pleasure, and I would like to use abseil also on these platforms.